kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/ai/inference/TensorDescriptor.java

```diff
@@ -0,0 +1,133 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ ******************************************************************************/
+package org.eclipse.kura.ai.inference;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * The TensorDescriptor class describes the {@link Tensor} used as input or output
+ * of a model for Artificial Intelligence and Machine Learning algorithms
+ *
+ * @since 2.3
+ */
+@ProviderType
+public class TensorDescriptor {
+
+    private final String name;
+    private final String type;
+    private Optional<String> format;
+    private final List<Long> shape;
+    private final Map<String, Object> parameters;
+
+    /**
+     * Instantiates a tensor descriptor
+     * 
+     * @param name
+     *            the name of the tensor
+     * @param type
+     *            a string representing the type of data contained in the tensor
+     * @param shape
+     *            the shape of the data
+     */
+    public TensorDescriptor(String name, String type, List<Long> shape) {
```

Comment:
Since we have a builder for ModelInfo, could it make sense to create a builder for this one as well and make it immutable?